Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -341,161 +342,224 @@ namespace ROS2::Utils
+                ancestorPaths.emplace_back(fileAncestorPath);
+            } while (fileAncestorPath != fileAncestorPath.RootPath());
+        }
+
+        // Loop through each prefix in the builder settings and attempt to resolve it as either an absolute path
+        // or a relative path that's relative in some way to the base file.
+        for (const auto& [prefix, replacements] : pathResolverSettings.m_uriPrefixMap)
         {
-            return [schemePrefix, amentPrefixPaths, urdfAncestorPaths, &fileExistsCB](
-                AZ::IO::PathView uriPath) -> AZStd::optional<AZ::IO::Path>
+            // Note this is a case-sensitive check to match the exact URI scheme
+            // If that is not desired, then this code should be updated to read
+            // a value from the Setting Registry indicating whether the uriPrefix matching
+            // should be case sensitive
+            constexpr bool uriPrefixMatchCaseSensitive = true;
+            // If the path doesn't start with the given prefix, move on to the next prefix
+            if (!AZ::StringFunc::StartsWith(unresolvedPath.Native(), prefix, uriPrefixMatchCaseSensitive))
             {
-                // Note this is a case-sensitive check to match the exact URI scheme
-                // If that is not desired, then this code should be updated to read
-                // a value from the Setting Registry indicating whether the uriPrefix matching
-                // should be case sensitive
-                bool uriPrefixMatchCaseSensitive = true;
-                // Check if the path starts with the URI scheme prefix
-                if (AZ::StringFunc::StartsWith(uriPath.Native(), schemePrefix, uriPrefixMatchCaseSensitive))
+                continue;
+            }
+
+            // Strip the number of characters from the Uri scheme from beginning of the path
+            AZ::IO::PathView strippedUriPath = AZ::IO::PathView(unresolvedPath).Native().substr(prefix.size());
+
+            // Loop through each replacement path for this prefix, attach it to the front, and look for matches.
+            for (const auto& replacement : replacements)
+            {
+                AZ::IO::Path replacedUriPath(replacement);
+                replacedUriPath /= strippedUriPath;
+
+                // If we successfully matched the prefix, and the replacement path is completely empty, we don't need to look any further.
+                // There's no match.
+                if (replacedUriPath.empty())
                 {
-                    // Strip the number of characters from the Uri scheme from beginning of the path
-                    AZ::IO::PathView strippedUriPath = uriPath.Native().substr(schemePrefix.size());
-                    if (strippedUriPath.empty())
+                    AZ_Trace("ResolveAssetPath", R"(Resolved Path is empty: "%.*s" -> "")" "\n", AZ_PATH_ARG(unresolvedPath));
+                    return "";
```

Comment:
This can return `{}` to use the default constructor.
It is slightly faster than initializing an AZ::IO::Path with an empty `const char*`, as a call to strlen still needs to occur internally to determine that the string is empty
```suggestion
                    return {};
```